fission-web-server/library/Fission/Web/Server/Internal/Tasks.hs

```diff
@@ -45,7 +54,71 @@ deleteByUsername userNameTxt =
         Nothing                -> error "User doesn't exist"
 
 ---
+syncDNS :: Server()
+syncDNS = do
+  users <- getUserData
+  apps <- getAppData
+  mapM_ setUserDNS users
+  mapM_ setAppDNS apps
+
+getUserData :: Server [Entity User]
+getUserData = do
+  runDB do
+    select $ from \user -> do
+      where_ $ SQL.isNothing (user ^. UserSecretDigest)
+      return user
 
+getAppData :: Server [(Entity App, Entity AppDomain)]
+getAppData = do
+  runDB do
+    select $ from \(app `InnerJoin` appDomain) -> do
+      SQL.on $ appDomain ^. AppDomainAppId ==. app ^. AppId
+      return (app, appDomain)
+
+setUserDNS :: Entity User -> Server()
+setUserDNS (Entity userId User { userUsername, userPublicKey, userDataRoot }) = do
+  zoneID <- asks userZoneID
+  domainName <- asks userRootDomain
+  let
+    unSubDom = Subdomain $ textDisplay userUsername
+    url      = URL {domainName, subdomain = Just (Subdomain "_did") <> Just unSubDom}
+    segments = case userPublicKey of
+      Nothing   -> pure ""
+      Just pkey -> DNS.splitRecord $ textDisplay (DID Key pkey)
+  PowerDNS.set PDNS.TXT url (textDisplay zoneID) segments 10 >>= \case
+    Left _ -> return ()
+    Right _ -> do
+      let
+        userUrl = URL
+          { domainName
+          , subdomain = Just . Subdomain $ textDisplay userUsername
+          }
+        userFilesUrl = URL
+          { domainName
+          , subdomain  = Just $ Subdomain (textDisplay userUsername  <> ".files")
+          }
+        userPublic = userFilesUrl `WithPath` ["public"]
+      DNSLink.follow userId userUrl zoneID userPublic >>= \case
+        Left _ -> return ()
+        Right _ -> do
+          DNSLink.set userId userFilesUrl zoneID userDataRoot >>= \case
+            Left _ -> return ()
+            Right _ ->
+              logDebug $ "Successfully updated " <> textDisplay userUsername
+
+setAppDNS :: (Entity App, Entity AppDomain) -> Server()
+setAppDNS (Entity _ App {appOwnerId, appCid}, Entity _ AppDomain {appDomainDomainName, appDomainSubdomain}) = do
+  zoneID     <- asks baseAppZoneID
+  let
+    url = URL
+      { domainName = appDomainDomainName
+      , subdomain = appDomainSubdomain
+      }
+  DNSLink.set appOwnerId url zoneID appCid >>= \case
+    Left _ -> return ()
+    Right _ ->
+      logDebug $ "Successfully updated " <> show appDomainSubdomain
+---
```

Comment:
Same here, feel free to add some space around the `---`. If you like C-style sections, you can also make pretty `--` boxes

```haskell
logDebug $ "Successfully updated " <> show appDomainSubdomain

---------------------
-- IPFS Migration --
---------------------

ensureAllPinned :: Server ()
```

(Though in many cases, that's a sign that you would want another module... but hey, this is helper scripts so _whatever_ 😛 )